libs/libdaemon/src/BaseDaemon.cpp

```diff
@@ -501,6 +502,34 @@ void BaseDaemon::closeFDs()
     }
 }
 
+namespace
+{
+/// In debug version on Linux, increase oom score so that clickhouse is killed
+/// first, instead of some service. Use a carefully chosen random score of 555:
+/// the maximum is 1000, and chromium uses 300 for its tab processes. Ignore
+/// whatever errors that occur, because it's just a debugging aid and we don't
+/// care if it breaks.
+#if defined(__linux__) and not defined(NDEBUG)
+void debugIncreaseOOMScore()
+{
+    const std::string new_score = "555";
+    try {
+        Poco::FileOutputStream file("/proc/self/oom_score_adj");
+        file << new_score;
```

Comment:
Does FileOutputStream throw exception when it cannot write to file. Or maybe it will just set ios flags?

Reply:
We can also use WriteBufferFromFile.

Reply:
Changed to WriteBufferFromFile.